### PR TITLE
[JENKINS-49737] - Issue(Print error to Java Version 9)

### PR DESCRIPTION
**Pull request by AmitJMagar**
I have simply followed the same logic which was followed for Java 7 or older

### CODE REVIEW

File: src/main/java/Main.java [hunk 1]
```diff
@@ -120,8 +120,10 @@ public static void main(String[] args) throws Exception {
             String v = System.getProperty("java.class.version");
             if (v!=null) {
                 try {
-                    if (Float.parseFloat(v)<52.0f)
+                	//Solving issue JENKINS-49737 adding warning for Java version 9
+                    if (Float.parseFloat(v)<52.0f || Float.parseFloat(v)==53.0f)
```
Comment (oleg-nenashev) [inline src/main/java/Main.java]: Java 10 is coming soon... Better to do an exact comparison

File: src/main/java/Main.java [hunk 2]
```diff
@@ -131,7 +133,8 @@ public static void main(String[] args) throws Exception {
 
             _main(args);
         } catch (UnsupportedClassVersionError e) {
-            System.err.println("Jenkins requires Java8 or later, but you are running "+
+        	// printing error for JENKINS-49737
+            System.err.println("Jenkins atleast requires Java8 and does not support Java 9, but you are running "+
```
Comment (oleg-nenashev) [inline src/main/java/Main.java]: Maybe just remove "or later"?

File: src/main/java/Main.java [hunk 3]
```diff
@@ -120,8 +120,10 @@ public static void main(String[] args) throws Exception {
             String v = System.getProperty("java.class.version");
             if (v!=null) {
                 try {
-                    if (Float.parseFloat(v)<52.0f)
+                	//Solving issue JENKINS-49737 adding warning for Java version 9
```
Comment (oleg-nenashev) [inline src/main/java/Main.java]: Such comments are not needed in general.

**Comment by oleg-nenashev** [comment]
The last version won't even compile...

**Comment by oleg-nenashev** [comment]
Thanks @AmitJMagar . I have fixed the last comments, the code is ready to go IMHO.
@jenkinsci/code-reviewers Any concerns? Esp. @batmat @MarkEWaite @daniel-beck

**Comment by oleg-nenashev** [comment]
#15 needs to be merged before the release

File: src/main/java/Main.java [hunk 4]
```diff
@@ -120,8 +120,8 @@ public static void main(String[] args) throws Exception {
             String v = System.getProperty("java.class.version");
             if (v!=null) {
                 try {
-                    if (Float.parseFloat(v)<52.0f)
-                        throw new UnsupportedClassVersionError(v);
+                    if (Float.parseFloat(v)!=52.0f)
```
Comment (batmat) [inline src/main/java/Main.java]: Shouldn't it rather be something like: 

```java
if( Math.abs(Float.parseFloat(v) - 52.0f) < 0.001f )
```
?

File: src/main/java/Main.java [hunk 5]
```diff
@@ -131,7 +131,7 @@ public static void main(String[] args) throws Exception {
 
             _main(args);
         } catch (UnsupportedClassVersionError e) {
-            System.err.println("Jenkins requires Java8 or later, but you are running "+
+            System.err.println("Jenkins requires Java8, but you are running "+
```
Comment (batmat) [inline src/main/java/Main.java]: NIT: I would take the opportunity to write `Java 8` with a space, not `Java8`.

File: src/main/java/Main.java [hunk 6]
```diff
@@ -120,8 +120,8 @@ public static void main(String[] args) throws Exception {
             String v = System.getProperty("java.class.version");
             if (v!=null) {
                 try {
-                    if (Float.parseFloat(v)<52.0f)
-                        throw new UnsupportedClassVersionError(v);
+                    if (Float.parseFloat(v)!=52.0f)
+                        throw new UnsupportedClassVersionError(v);
```
Comment (batmat) [inline src/main/java/Main.java]: OMG, not your change I know, but well, we really throw this ourselves?... All my basics are being challenged :)

File: src/main/java/Main.java (hunk 4 above)
Comment (oleg-nenashev) [reply src/main/java/Main.java]: Should not matter. Float comparison is safe enough due to internal margins. Well, it weird but safe

**Comment by AmitJMagar** [comment]
@oleg-nenashev @batmat Thank you guys for reviewing it this is my first involvement in open source,

**Comment by AmitJMagar** [comment]
@oleg-nenashev  Saw my mistake, I got it why the code wasn't compiling, I was making changes in github website only hence couldn't catch that error, Next time I will double check

**Comment by oleg-nenashev** [comment]
@AmitJMagar OK, now we have CI in place after #15 . I will integrate and release it

**Pull request merged**